packages/renderer/src/lib/preferences/PreferencesCliTool.svelte

```diff
@@ -0,0 +1,135 @@
+<script lang="ts">
+import { faCircleArrowUp, faCircleXmark } from '@fortawesome/free-solid-svg-icons';
+import Button from '../ui/Button.svelte';
+import Fa from 'svelte-fa';
+import { startTask, type ConnectionCallback, eventCollect } from './preferences-connection-rendering-task';
+import type { CliToolInfo } from '../../../../main/src/plugin/api/cli-tool-info';
+import LoadingIconButton from '../ui/LoadingIconButton.svelte';
+
+import Markdown from '../markdown/Markdown.svelte';
+import type { ILoadingStatus } from './Util';
+
+export let cliTool: CliToolInfo;
+let showError = false;
+let cliToolStatus: ILoadingStatus = {
+  inProgress: false,
+  status: cliTool.newVersion ? 'toUpdate' : 'unknown',
+  action: 'update',
+};
+
+async function update(cliTool: CliToolInfo) {
+  try {
+    cliToolStatus.inProgress = true;
+    cliToolStatus = cliToolStatus;
+    const loggerHandlerKey = startTask(
+      `Update ${cliTool.name} to v${cliTool.newVersion}`,
+      '/preferences/cli-tools',
+      getLoggerHandler(cliTool.id),
+    );
+    await window.updateCliTool(cliTool.id, loggerHandlerKey, eventCollect);
+    showError = false;
+    cliToolStatus.status = 'unknown';
+  } catch (e) {
+    showError = true;
+  } finally {
+    cliToolStatus.inProgress = false;
+    cliToolStatus = cliToolStatus;
+  }
+}
+
+function getLoggerHandler(_cliToolId: string): ConnectionCallback {
+  return {
+    log: () => {},
+    warn: () => {},
+    error: _args => {
+      showError = true;
+    },
+    onEnd: () => {},
+  };
+}
+</script>
+
+<div role="row" class="bg-charcoal-600 mb-5 rounded-md p-3 flex flex-col">
+  <div class="divide-x divide-gray-900 flex flex-row">
+    <div>
+      <!-- left col - cli-tool icon/name + "create new" button -->
+      <div class="min-w-[170px] max-w-[200px] h-full flex flex-col justify-between">
+        <div class="flex flex-row">
+          {#if cliTool?.images?.icon}
```

Comment:
should it display the extension icon if there is none on the clitool part ?

let say for kubectl extension I would assume that we can use extension icon

Reply:
it could make sense. I wonder if we should make the image mandatory so the extension is forced to provide an icon or we should ask for additional extensionInfo